crates/lib/src/qpu/quilc/mod.rs

```diff
@@ -98,7 +97,7 @@ impl AsRef<str> for NativeQuil {
 /// A wrapper around [`Program`] which indicates it has been converted to `NativeQuil` (has been run
 /// through `quilc` and therefore is suitable to use on QPUs.
 #[derive(Debug, PartialEq, Clone)]
-pub(super) struct NativeQuilProgram(Program);
```

Comment:
I don't understand why this struct is needed - a program is a program, and what is "native" is actually up to the remote API to determine. Errors should make it clear when a program is native or not, and so I think this type wrapper adds noise rather than signal.

Reply:
Agreed. It was added long ago (unrelated to this MR or the Python bindings MRs) but I'll remove it anyhow.